backbone/src/main/java/org/researchstack/backbone/ui/task/TaskViewModel.kt

```diff
@@ -280,4 +278,47 @@ internal class TaskViewModel(val context: Application, intent: Intent) : Android
         stack.clear()
         editing = false
     }
+
+    fun checkForSaveDialog(stepResult: StepResult<*>?) {
+        if ((!isSavedDialogAppeared && !checkIfAnswersAreTheSame(stepResult) && checkIfCurrentStepIsBranchDecisionStep()) ||
+                (currentStep!!.isOptional && checkIfTheNewAnswerIsSkipWhileThePreviousIsNot(stepResult))) {
+            isSavedDialogAppeared = true
+            showSaveEditDialog.postValue(true)
+        } else {
+            nextStep()
+        }
+    }
+
+    fun saveEditDialogDismiss() {
+        isSavedDialogAppeared = false
+    }
+
+    private fun checkIfAnswersAreTheSame(currentAnswer: StepResult<*>?): Boolean {
+        return clonedTaskResultInCaseOfCancel?.let {
+            val previousAnswer = it.getStepAndResult(currentStep!!.identifier).second
+            return previousAnswer?.let { previousAnswerResult ->
+                previousAnswerResult.result == currentAnswer?.let { currentAnswerResult ->
+                    currentAnswerResult.result
+                } ?: false
+            } ?: false
+        } ?: false
+    }
+
+
+    private fun checkIfTheNewAnswerIsSkipWhileThePreviousIsNot(currentAnswer: StepResult<*>?): Boolean {
```

Comment:
Do you mind renaming this to: 

`checkIfNewAnswerIsSkipWhilePreviousIsNot(newAnswer: StepResult<*>?): Boolean {`

I believe the wording of "New Answer" in the method name, but then the parameter being "currentAnswer" can cause confusion to the reader, because is not clear if "current" is "new" (it is in this context)

What do you think?